src/cookbooks/main_ping_exponential_histograms.md

```diff
@@ -0,0 +1,312 @@
+
+## Creating a time series of change over time
+
+Knowing the approximate distribution of results on a given day is sort of interesting, but probably not what we really want: what we're usually interested in is the evolution of results _over time_. In particular, segmenting by `build_id` (a date-like structure in the main ping representing when Firefox was built) is a useful technique, as it allows us to see if changes to Firefox itself may have caused the distribution to change.
+
+We can do this simply by _grouping by_ the build id field, and then merging the histograms corresponding to each:
+
+```sql
+DECLARE four_twenty DEFAULT DATE('2020-04-20');
+
+WITH per_build_day AS
+  (SELECT PARSE_DATETIME("%Y%m%d%H%M%S", application.build_id) AS build_id,
+          KEY,
+          SUM(value) AS value,
+   FROM `moz-fx-data-shared-prod`.telemetry.main,
+        UNNEST(`moz-fx-data-shared-prod`.udf.json_extract_histogram(payload.histograms.FX_TAB_SWITCH_SPINNER_VISIBLE_MS).
+               VALUES)
+   WHERE application.channel='nightly'
+     AND normalized_os='Windows'
+     AND application.build_id > FORMAT_DATE("%Y%m%d", DATE_SUB(four_twenty, INTERVAL 2 WEEK))
+     AND application.build_id <= FORMAT_DATE("%Y%m%d", four_twenty)
+     AND DATE(submission_timestamp) >= DATE_SUB(four_twenty, INTERVAL 2 WEEK)
+     AND DATE(submission_timestamp) <= four_twenty
+   GROUP BY KEY,
+            build_id),
+     per_build_day_as_struct AS
+  (SELECT build_id,
+          STRUCT(ARRAY_AGG(STRUCT(KEY, value)) AS
+                 VALUES) AS spinner_visible_ms
+   FROM per_build_day
+   GROUP BY build_id)
+
+SELECT build_id,
+       percentile,
+       value
+FROM per_build_day_as_struct
+CROSS JOIN UNNEST (`moz-fx-data-shared-prod`.udf.histogram_percentiles(spinner_visible_ms, [.05,
+      .25,
+      .5,
+      .75,
+      .95]))
+```
+
+[link](https://sql.telemetry.mozilla.org/queries/71472/source)
+
+As an implementation note, observe that we don't use `histogram_merge` here as we do above: doing so would require using [`ARRAY_AGG`](https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#array_agg) which can break down when processing large amounts of data. Instead we create a temporary table (`per_build_day`) by hand and then reprocess it into a structured representation. If you're curious what the version using `histogram_merge` would look like, see [this example](https://sql.telemetry.mozilla.org/queries/71413/source).
+
+In any case, rendering the data this query returns, we get a chart like this:
+
+![](https://i.imgur.com/HByDexv.png)
```

Comment:
I will change these to be static assets once we've verified for certain they're what we want.